### PR TITLE
Blog App - processing data in models

### PR DESCRIPTION
**Pull request by usmansbk**
# Project 4: Processing data in models.

In this PR, I did the following:

- Generate models
- Create fake authentication
- Create associations
- Create model methods
- Add automatic counters

### CODE REVIEW

File: app/models/comment.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post, counter_cache: :comments_counter
+end
```
Comment (de-scholar) [inline app/models/comment.rb]: - [ ] Comment model should have a method that updates the comments counter for a post so would you add this method as well?

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post, counter_cache: :likes_counter
+end
```
Comment (de-scholar) [inline app/models/like.rb]: - [ ] Like model should have a method that updates the likes counter for a post so would you add this method as well?

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+class Post < ApplicationRecord
+  belongs_to :user, counter_cache: :posts_counter
+  has_many :comments, dependent: :destroy
+  has_many :likes, dependent: :destroy
+
+  def recent_comments(limit = 5)
+    comments.last(limit)
+  end
+end
```
Comment (de-scholar) [inline app/models/post.rb]: - [ ] Post model should a method that updates the posts counter for a user so would you add this method as well?

**Review by de-scholar — CHANGES_REQUESTED** [review]
Hi @usmansbk , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 3 above)
Comment (usmansbk) [reply app/models/post.rb]: Well the counters update automatically thanks to the Rails counter_cache. They say we can improve on things that we believe is better.

```rb
  belongs_to :user, counter_cache: :posts_counter
```

Comment (usmansbk) [reply app/models/post.rb]: Try creating and deleting models in Rails console and see it in effect

Comment (usmansbk) [reply app/models/post.rb]: @de-scholar

**Comment by usmansbk** [comment]
@de-scholar 

> Use what you have learned so far, especially best practices and refactor your app to make it faster, more maintainable, or whatever improvement you see fit.

**Review by AshakaE — APPROVED** [review]
### Hi @usmansbk  🙋‍♀️, 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**